libraries/AP_RangeFinder/AP_RangeFinder_Benewake_CAN.cpp

```diff
@@ -0,0 +1,74 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AP_RangeFinder_Benewake_CAN.h"
+
+#if HAL_MAX_CAN_PROTOCOL_DRIVERS
+
+const AP_Param::GroupInfo AP_RangeFinder_Benewake_CAN::var_info[] = {
+
+    // @Param: RECV_ID
+    // @DisplayName: CAN receive ID
+    // @Description: The receive ID of the CAN frames. A value of zero means all IDs are accepted.
+    // @Range: 0 65535
+    // @User: Advanced
+    AP_GROUPINFO("RECV_ID", 10, AP_RangeFinder_Benewake_CAN, receive_id, 0),
+
+    // @Param: SNR_MIN
+    // @DisplayName: Minimum signal strength
+    // @Description: Minimum signal strength (SNR) to accept distance
+    // @Range: 0 65535
+    // @User: Advanced
+    AP_GROUPINFO("SNR_MIN", 11, AP_RangeFinder_Benewake_CAN, snr_min, 0),
+
+    AP_GROUPEND
+};
+
+
+/*
+  constructor
+ */
+AP_RangeFinder_Benewake_CAN::AP_RangeFinder_Benewake_CAN(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params) :
+    CANSensor("Benewake"),
+    AP_RangeFinder_Backend(_state, _params)
+{
+    register_driver(AP_CANManager::Driver_Type_Benewake);
+    AP_Param::setup_object_defaults(this, var_info);
```

Comment:
I'm a bit surprised to see these around this way.

```suggestion
    AP_Param::setup_object_defaults(this, var_info);
    register_driver(AP_CANManager::Driver_Type_Benewake);
```

Reply:
done